mirrord/protocol/src/features.rs

```diff
@@ -0,0 +1,194 @@
+use std::{cell::RefCell, fmt, str::FromStr};
+
+use bitflags::bitflags;
+use bitflags_serde_shim::impl_serde_for_bitflags;
+use serde::{Deserialize, Serialize};
+
+bitflags! {
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+    pub struct Features: u32 { }
+}
+
+impl_serde_for_bitflags!(Features);
+
+impl fmt::Display for Features {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+impl FromStr for Features {
+    type Err = bitflags::parser::ParseError;
+
+    fn from_str(flags: &str) -> Result<Self, Self::Err> {
+        Ok(Self(flags.parse()?))
+    }
+}
+
+thread_local!(
```

Comment:
This is ugly, what can we do about it?
I'd prefer it to be a OnceLock/LazyCell - no need for thread local..